sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileSmbProperties.java

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file;
+
+import com.azure.storage.file.models.DirectorysCreateResponse;
+import com.azure.storage.file.models.DirectorysGetPropertiesResponse;
+import com.azure.storage.file.models.DirectorysSetPropertiesResponse;
+import com.azure.storage.file.models.FilesCreateResponse;
+import com.azure.storage.file.models.FilesDownloadResponse;
+import com.azure.storage.file.models.FilesGetPropertiesResponse;
+import com.azure.storage.file.models.FilesSetHTTPHeadersResponse;
+import com.azure.storage.file.models.NtfsFileAttributes;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.EnumSet;
+
+public class FileSmbProperties {
```

Comment:
Would it make sense to split this into two classes, one for request data and another for response data?

Reply:
I think it would be convenient to keep them in the same class due to the convenience of the user being able to just extract the FileSMBProperties and change what they want if they want to ever re set them. But I'm open to the idea. Anyone else have any major issues with one class?